File: Server/ExchangeRates.Server/Models/CurrencyModel.cs
namespace ExchangeRates.Server.Models;

public sealed class CurrencyModel {
	public CurrencyModel(string currencyCode, int numericCode, string name, bool isHistoric) {
		CurrencyCode = currencyCode;
		Name = name;
		IsHistoric = isHistoric;
		NumericCode = numericCode;
	}

	public string CurrencyCode { get; }
	public string Name { get; }
	public bool IsHistoric { get; }
	public int NumericCode { get; }
}

File: Clients/ExchangeRates.Client/exchangerates.client.react/src/models/CurrencyModel.js
export default class CurrencyModel {
    constructor(currencyCode, numericCode, name, isHistoric) {
        this.currencyCode = String(currencyCode);
        this.numericCode = Number(numericCode);
        this.name = String(name);
        this.isHistoric = Boolean(isHistoric);
    }
}